### PR TITLE
Update ztunnel documentation to use ztunnel-config command instead of…

### PR DESCRIPTION
**Pull request by MorrisLaw**
## Description
This PR updates references to `istioctl proxy-config` to be `istioctl x ztunnel-config` instead, since recent istioctl changes have changed the way ztunnel configuration info can be retrieved. `istioctl proxy-config` no longer works for getting this info.

Related to #14983

## Reviewers

- [x] Ambient
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

### CODE REVIEW

**Comment by MorrisLaw** [comment]
cc/ @linsun @bleggett

File: content/en/docs/ambient/usage/ztunnel/index.md [hunk 1]
```diff
@@ -226,7 +226,7 @@ $ echo "$ZTUNNEL"
 {{< /text >}}
 
 {{< text bash >}}
-$ istioctl proxy-config workloads "$ZTUNNEL".istio-system
+$ istioctl x ztunnel-config workloads "$ZTUNNEL".istio-system
```
Comment (linsun) [inline content/en/docs/ambient/usage/ztunnel/index.md:229]: Does user needs to specify "$ZTUNNEL".istio-system?  I don't think it is required.

File: content/en/docs/ambient/usage/ztunnel/index.md [hunk 2]
```diff
@@ -226,7 +226,7 @@ $ echo "$ZTUNNEL"
...
 NAME                                   NAMESPACE          IP         NODE               WAYPOINT PROTOCOL
 coredns-6d4b75cb6d-ptbhb               kube-system        10.240.0.2 amb1-control-plane None     TCP
 coredns-6d4b75cb6d-tv5nz               kube-system        10.240.0.3 amb1-control-plane None     TCP
```
Comment (linsun) [inline content/en/docs/ambient/usage/ztunnel/index.md:232]: The outpit needs update i think?

File: content/en/docs/ambient/usage/ztunnel/index.md [hunk 3]
```diff
@@ -245,7 +245,7 @@ ztunnel-xxbgj                          istio-system       10.240.2.2 amb1-worker
 In the second example, you see the list of TLS certificates that this ztunnel proxy instance has received from istiod to use in TLS signaling.
 
 {{< text bash >}}
-$ istioctl proxy-config secrets "$ZTUNNEL".istio-system
+$ istioctl x ztunnel-config certificates "$ZTUNNEL".istio-system
```
Comment (linsun) [inline content/en/docs/ambient/usage/ztunnel/index.md:248]: same comment here.  also output needs to be updated

File: content/en/docs/ambient/usage/ztunnel/index.md [hunk 4]
```diff
@@ -257,13 +257,13 @@ spiffe://cluster.local/ns/istio-system/sa/ztunnel     CA             Available
...
 {{< /text >}}
 
-Using these CLI commands, a user can check that ztunnel proxies are getting configured with all the expected workloads and TLS certificates and missing information can be used for troubleshooting to explain any potential observed networking errors. A user may also use the `all` option to view all parts of the proxy-config with a single CLI command and the JSON output formatter as shown in the example below to display the complete set of available state information.
+Using these CLI commands, a user can check that ztunnel proxies are getting configured with all the expected workloads and TLS certificates and missing information can be used for troubleshooting to explain any potential observed networking errors. A user may also use the `all` option to view all parts of the ztunnel-config with a single CLI command and the JSON output formatter as shown in the example below to display the complete set of available state information.
 
 {{< text bash >}}
-$ istioctl proxy-config all "$ZTUNNEL".istio-system -o json | jq
+$ istioctl x ztunnel-config all "$ZTUNNEL".istio-system -o json | jq
```
Comment (linsun) [inline content/en/docs/ambient/usage/ztunnel/index.md:263]: same comment here

Comment (linsun) [reply content/en/docs/ambient/usage/ztunnel/index.md:263]: check if jq is required too?

File: content/en/docs/ambient/usage/ztunnel/index.md (hunk 1 above)
Comment (MorrisLaw) [reply content/en/docs/ambient/usage/ztunnel/index.md:229]: I don't think it's required, I can verify though. I was just going off of previous functionality which still seems to work.

Comment (linsun) [reply content/en/docs/ambient/usage/ztunnel/index.md:229]: Let us keep the doc as simple as possible - It is not required so let us remove it?

Comment (MorrisLaw) [reply content/en/docs/ambient/usage/ztunnel/index.md:229]: agreed!

Comment (ericvn) [reply content/en/docs/ambient/usage/ztunnel/index.md:229]: We also expand out x to experimental

Comment (craigbox) [reply content/en/docs/ambient/usage/ztunnel/index.md:229]: (I do wonder if the UX of having `istioctl x` is somewhat at odds with how we use feature statuses, and should be removed.  I'll bring this up on Slack)

File: content/en/docs/ambient/usage/ztunnel/index.md (hunk 4 above)
Comment (MorrisLaw) [reply content/en/docs/ambient/usage/ztunnel/index.md]: `jq` isn't required

**Comment by MorrisLaw** [comment]
> requested some changes see comments

markdown seems off in the commit, I'll preview it then fix it if necessary

File: content/en/docs/ambient/usage/ztunnel/index.md [hunk 5]
```diff
@@ -135,55 +135,45 @@ This section describes some options for monitoring the ztunnel proxy configurati
 
 ### Viewing ztunnel proxy state
 
-As indicated previously, the ztunnel proxy on each node gets configuration and discovery information from the istiod component via xDS APIs. Use the `istioctl proxy-config` command shown below to view discovered workloads as seen by a ztunnel proxy as well as secrets holding the TLS certificates that the ztunnel proxy has received from the istiod control plane to use in mTLS signaling on behalf of the local workloads.
+As indicated previously, the ztunnel proxy on each node gets configuration and discovery information from the istiod component via xDS APIs. Use the `istioctl experimental ztunnel-config` command shown below to view discovered workloads as seen by a ztunnel proxy as well as secrets holding the TLS certificates that the ztunnel proxy has received from the istiod control plane to use in mTLS signaling on behalf of the local workloads.
```
Comment (ilrudie) [inline content/en/docs/ambient/usage/ztunnel/index.md:138]:
```suggestion
As indicated previously, the ztunnel proxy on each node gets configuration and discovery information from the istiod component via xDS APIs. Use the `istioctl experimental ztunnel-config` command shown below to view discovered workloads as seen by a ztunnel proxy as well as the TLS certificates that the ztunnel proxy has received from the istiod control plane to use in mTLS signaling on behalf of the local workloads.
```

File: content/en/docs/ambient/usage/ztunnel/index.md [hunk 6]
```diff
@@ -135,55 +135,45 @@ This section describes some options for monitoring the ztunnel proxy configurati
...
+spiffe://cluster.local/ns/ambient-demo/sa/sleep        Root     Available     true           b73bb47b4801d9c0b4d13e5cc79afab7     2034-05-01T17:50:10Z     2024-05-03T17:50:10Z
 {{< /text >}}
 
-Using these CLI commands, a user can check that ztunnel proxies are getting configured with all the expected workloads and TLS certificates and missing information can be used for troubleshooting to explain any potential observed networking errors. A user may also use the `all` option to view all parts of the proxy-config with a single CLI command and the JSON output formatter as shown in the example below to display the complete set of available state information.
+Using these CLI commands, a user can check that ztunnel proxies are getting configured with all the expected workloads and TLS certificates and missing information can be used for troubleshooting to explain any potential observed networking errors. A user may also use the `all` option to view all parts of the ztunnel-config with a single CLI command and the JSON output formatter as shown in the example below to display the complete set of available state information.
```
Comment (ilrudie) [inline content/en/docs/ambient/usage/ztunnel/index.md:171]:
```suggestion
Using these CLI commands, a user can check that ztunnel proxies are getting configured with all the expected workloads and TLS certificates. Additionally, missing information may be relevant for troubleshooting to explain observed networking errors. A user may also use the `all` option to view all parts of the ztunnel-config with a single CLI command and the JSON output formatter as shown in the example below to display the complete set of available state information.
```

**Review by ilrudie — COMMENTED** [review]
Just a few minor suggestions for concise, clear wording.

**Comment by craigbox** [comment]
superseded by #15028; I've taken the changes from this PR that I hadn't found out by trial and error (hint: review the queue before doing anything) and included them.

However, we need to double check: I get certificate output when I specify the ztunnel pod name, but not when I leave it off, as this example does.

```
$ export ZTUNNEL=$(kubectl get pods -n istio-system -l app=ztunnel -o=jsonpath='{.items[0].metadata.name}')
$ istioctl x zc certificates
CERTIFICATE NAME     TYPE     STATUS     VALID CERT     SERIAL NUMBER     NOT AFTER     NOT BEFORE
$ istioctl x zc certificates "$ZTUNNEL".istio-system
CERTIFICATE NAME                                              TYPE     STATUS        VALID CERT     SERIAL NUMBER                        NOT AFTER                NOT BEFORE
spiffe://cluster.local/ns/default/sa/bookinfo-details         Leaf     Available     false          c198d859ee51556d0eae13b331b0c259     2024-05-05T09:17:47Z     2024-05-04T09:15:47Z
spiffe://cluster.local/ns/default/sa/bookinfo-details         Root     Available     true           bad086c516cce777645363cb8d731277     2034-04-24T03:31:05Z     2024-04-26T03:31:05Z
spiffe://cluster.local/ns/default/sa/bookinfo-productpage     Leaf     Available     false          64c3828993c7df6f85a601a1615532cc     2024-05-05T09:17:47Z     2024-05-04T09:15:47Z
spiffe://cluster.local/ns/default/sa/bookinfo-productpage     Root     Available     true           bad086c516cce777645363cb8d731277     2034-04-24T03:31:05Z     2024-04-26T03:31:05Z
spiffe://cluster.local/ns/default/sa/bookinfo-ratings         Leaf     Available     false          720479815bf6d81a05df8a64f384ebb0     2024-05-05T09:17:47Z     2024-05-04T09:15:47Z
spiffe://cluster.local/ns/default/sa/bookinfo-ratings         Root     Available     true           bad086c516cce777645363cb8d731277     2034-04-24T03:31:05Z     2024-04-26T03:31:05Z
spiffe://cluster.local/ns/default/sa/bookinfo-reviews         Leaf     Available     true           54f1ba49500da5803e77dc9e874edcdb     2024-05-05T10:16:53Z     2024-05-04T10:14:53Z
spiffe://cluster.local/ns/default/sa/bookinfo-reviews         Root     Available     true           bad086c516cce777645363cb8d731277     2034-04-24T03:31:05Z     2024-04-26T03:31:05Z
spiffe://cluster.local/ns/default/sa/sleep                    Leaf     Available     false          fa33bbb783553a1704866842586e4c0b     2024-05-05T09:25:49Z     2024-05-04T09:23:49Z
spiffe://cluster.local/ns/default/sa/sleep                    Root     Available     true           bad086c516cce777645363cb8d731277     2034-04-24T03:31:05Z     2024-04-26T03:31:05Z
```

**Comment by MorrisLaw** [comment]
> superseded by #15028; I've taken the changes from this PR that I hadn't found out by trial and error (hint: review the queue before doing anything) and included them.
> 
> However, we need to double check: I get certificate output when I specify the ztunnel pod name, but not when I leave it off, as this example does.
> 
> ```
> $ export ZTUNNEL=$(kubectl get pods -n istio-system -l app=ztunnel -o=jsonpath='{.items[0].metadata.name}')
> $ istioctl x zc certificates
> CERTIFICATE NAME     TYPE     STATUS     VALID CERT     SERIAL NUMBER     NOT AFTER     NOT BEFORE
> $ istioctl x zc certificates "$ZTUNNEL".istio-system
> CERTIFICATE NAME                                              TYPE     STATUS        VALID CERT     SERIAL NUMBER                        NOT AFTER                NOT BEFORE
> spiffe://cluster.local/ns/default/sa/bookinfo-details         Leaf     Available     false          c198d859ee51556d0eae13b331b0c259     2024-05-05T09:17:47Z     2024-05-04T09:15:47Z
> spiffe://cluster.local/ns/default/sa/bookinfo-details         Root     Available     true           bad086c516cce777645363cb8d731277     2034-04-24T03:31:05Z     2024-04-26T03:31:05Z
> spiffe://cluster.local/ns/default/sa/bookinfo-productpage     Leaf     Available     false          64c3828993c7df6f85a601a1615532cc     2024-05-05T09:17:47Z     2024-05-04T09:15:47Z
> spiffe://cluster.local/ns/default/sa/bookinfo-productpage     Root     Available     true           bad086c516cce777645363cb8d731277     2034-04-24T03:31:05Z     2024-04-26T03:31:05Z
> spiffe://cluster.local/ns/default/sa/bookinfo-ratings         Leaf     Available     false          720479815bf6d81a05df8a64f384ebb0     2024-05-05T09:17:47Z     2024-05-04T09:15:47Z
> spiffe://cluster.local/ns/default/sa/bookinfo-ratings         Root     Available     true           bad086c516cce777645363cb8d731277     2034-04-24T03:31:05Z     2024-04-26T03:31:05Z
> spiffe://cluster.local/ns/default/sa/bookinfo-reviews         Leaf     Available     true           54f1ba49500da5803e77dc9e874edcdb     2024-05-05T10:16:53Z     2024-05-04T10:14:53Z
> spiffe://cluster.local/ns/default/sa/bookinfo-reviews         Root     Available     true           bad086c516cce777645363cb8d731277     2034-04-24T03:31:05Z     2024-04-26T03:31:05Z
> spiffe://cluster.local/ns/default/sa/sleep                    Leaf     Available     false          fa33bbb783553a1704866842586e4c0b     2024-05-05T09:25:49Z     2024-05-04T09:23:49Z
> spiffe://cluster.local/ns/default/sa/sleep                    Root     Available     true           bad086c516cce777645363cb8d731277     2034-04-24T03:31:05Z     2024-04-26T03:31:05Z
> ```

This is what I am getting. I get results for both, but I am not sure why yours would be different. @linsun if you can confirm?

```
azureuser@dev-2024:~/istio$ k get po -A | grep ztunnel
istio-system         ztunnel-g9hzr                                         1/1     Running   0          10m

azureuser@dev-2024:~/istio$ export ZTUNNEL=$(kubectl get pods -n istio-system -o wide | grep ztunnel -m 1 | sed 's/ .*//')

azureuser@dev-2024:~/istio$ echo "$ZTUNNEL"
ztunnel-g9hzr

azureuser@dev-2024:~/istio$ ik x zc certificates 
CERTIFICATE NAME      TYPE     STATUS        VALID CERT     SERIAL NUMBER                        NOT AFTER                NOT BEFORE               
spiffe://cluster.local/ns/ambient-demo/sa/httpbin      Leaf     Available     true           a89260d272f6ed1fa9e7dfdf809a04a9     2024-05-07T15:41:06Z     2024-05-06T15:39:06Z
spiffe://cluster.local/ns/ambient-demo/sa/httpbin      Root     Available     true           6c1da936907ddce9594b85725d18ced6     2034-05-04T15:31:26Z     2024-05-06T15:31:26Z
spiffe://cluster.local/ns/ambient-demo/sa/notsleep     Leaf     Available     true           9d549ca6b291f78932ec14408fb18b35     2024-05-07T15:41:06Z     2024-05-06T15:39:06Z
spiffe://cluster.local/ns/ambient-demo/sa/notsleep     Root     Available     true           6c1da936907ddce9594b85725d18ced6     2034-05-04T15:31:26Z     2024-05-06T15:31:26Z
spiffe://cluster.local/ns/ambient-demo/sa/sleep        Leaf     Available     true           6ac206a87327e6feda0bdbb371350dd8     2024-05-07T15:41:06Z     2024-05-06T15:39:06Z
spiffe://cluster.local/ns/ambient-demo/sa/sleep        Root     Available     true           6c1da936907ddce9594b85725d18ced6     2034-05-04T15:31:26Z     2024-05-06T15:31:26Z

azureuser@dev-2024:~/istio$ ik x zc certificates "$ZTUNNEL".istio-system
CERTIFICATE NAME      TYPE     STATUS        VALID CERT     SERIAL NUMBER                        NOT AFTER                NOT BEFORE
spiffe://cluster.local/ns/ambient-demo/sa/httpbin      Leaf     Available     true           a89260d272f6ed1fa9e7dfdf809a04a9     2024-05-07T15:41:06Z     2024-05-06T15:39:06Z
spiffe://cluster.local/ns/ambient-demo/sa/httpbin      Root     Available     true           6c1da936907ddce9594b85725d18ced6     2034-05-04T15:31:26Z     2024-05-06T15:31:26Z
spiffe://cluster.local/ns/ambient-demo/sa/notsleep     Leaf     Available     true           9d549ca6b291f78932ec14408fb18b35     2024-05-07T15:41:06Z     2024-05-06T15:39:06Z
spiffe://cluster.local/ns/ambient-demo/sa/notsleep     Root     Available     true           6c1da936907ddce9594b85725d18ced6     2034-05-04T15:31:26Z     2024-05-06T15:31:26Z
spiffe://cluster.local/ns/ambient-demo/sa/sleep        Leaf     Available     true           6ac206a87327e6feda0bdbb371350dd8     2024-05-07T15:41:06Z     2024-05-06T15:39:06Z
spiffe://cluster.local/ns/ambient-demo/sa/sleep        Root     Available     true           6c1da936907ddce9594b85725d18ced6     2034-05-04T15:31:26Z     2024-05-06T15:31:26Z 
```

FWIW, these should both be valid commands according to istioctl:

```
azureuser@dev-2024:~/istio$ ik x zc certificate --help
Retrieve information about certificates for the Ztunnel instance.

Usage:
    istioctl experimental ztunnel-config certificate [flags]

Aliases:
    certificate, certificates, certs, cert
    
Examples:
    # Retrieve summary about workload configuration for a randomly chosen ztunnel.
    istioctl x ztunnel-config certificates

    # Retrieve full certificate dump of workloads for a given Ztunnel instance.
    istioctl x ztunnel-config certificates <ztunnel-name[.namespace]> -o json      
```

**Comment by MorrisLaw** [comment]
Superceded by #15028

**Pull request closed without merging**